scripts/install-dependencies.js

```diff
@@ -98,63 +100,71 @@ const installConfiguration = {
       platforms: ["win32-x64"],
     },
   ],
-  variables: [
-    {
-      platforms: ["linux-x64", "alpine-x64"],
+
+  platforms: {
+    "linux-x64": {
+      platform_aliases: ["alpine-x64"],
       os: "Linux",
       arch: "x86-64",
       package_dependencies_platform: "linux-x64",
       vectordb_platform: "linux-x64-gnu",
       esbuild_platform: "linux-64",
     },
-    {
-      platforms: ["linux-arm64", "alpine-arm64"],
+    "linux-arm64": {
+      platform_aliases: ["alpine-arm64"],
       os: "Linux",
       arch: "arm64",
       package_dependencies_platform: "linux-arm64",
       vectordb_platform: "linux-arm64-gnu",
       esbuild_platform: "linux-arm64",
     },
-    {
-      platforms: ["darwin-arm64"],
+    "darwin-arm64": {
       os: "macOS",
       arch: "arm64",
       package_dependencies_platform: "darwin-arm64",
       vectordb_platform: "darwin-arm64",
       esbuild_platform: "darwin-arm64",
     },
-    {
-      platforms: ["win32-x64"],
+    "win32-x64": {
       os: "Windows",
       arch: "x86-64",
       package_dependencies_platform: "win32-x64",
       vectordb_platform: "win32-x64-msvc",
       esbuild_platform: "windows-64",
     },
-  ],
+  },
   root: "./out",
 };
 
-function expandInstallConfigurationEntry(entry, variables) {
-  const expandedEntries = [];
-
-  if (!variables) return [entry];
+function expandInstallConfigurationEntry(entry, platforms, target) {
+  if (!platforms) return [entry];
 
-  for (const variableEntries of variables) {
-    if (
-      entry.platforms &&
-      !entry.platforms.some((p) => variableEntries.platforms.includes(p))
-    ) {
-      continue;
+  const expandedEntries = [];
+  const relevantPlatforms = target
+    ? Object.entries(platforms).filter(([platform, variables]) =>
+        [platform, ...(variables.platform_aliases || [])].some((platformName) =>
+          minimatch(target, platformName),
```

Comment:
Should this just be '==' - it's weird to support wildcards in platform_aliases. I guess it's consistent with `create-package.js`